corehq/apps/locations/views.py

```diff
@@ -100,37 +99,59 @@ def page_context(self):
     @property
     def settings_context(self):
         return {
-            'loc_types': [self._get_loctype_info(l) for l in self.domain_object.location_types],
+            'loc_types': map(
+                self._get_loctype_info,
+                LocationType.objects.filter(domain=self.domain).all()
+            )
         }
 
     def _get_loctype_info(self, loctype):
         return {
             'name': loctype.name,
             'code': loctype.code,
-            'allowed_parents': [p or None for p in loctype.allowed_parents],
+            'allowed_parents': [loctype.parent_type.code
+                                if loctype.parent_type else None],
             'administrative': loctype.administrative,
         }
 
     def post(self, request, *args, **kwargs):
         payload = json.loads(request.POST.get('json'))
 
-        def mk_loctype(loctype):
-            loctype['allowed_parents'] = [p or '' for p in loctype['allowed_parents']]
-            cleaned_code = unicode_slug(loctype['code'])
-            if cleaned_code != loctype['code']:
-                err = _(
-                    'Location type code "{code}" is invalid. No spaces or special characters are allowed. '
-                    'It has been replaced with "{new_code}".'
+        def mk_loctype(name, code, allowed_parents, administrative):
+            parents = allowed_parents
+            if len(parents) > 1:
+                messages.warning(request, _("Location types cannot have "
```

Comment:
F823 local variable '_' (defined in enclosing scope on line 11) referenced before assignment